site/app/libraries/homework/Gateways/LibraryGateway.php

```diff
@@ -0,0 +1,43 @@
+<?php namespace app\libraries\homework\Gateways;
+
+
+interface LibraryGateway {
+    /**
+     * This will clone the provided repo url into the specified location.
+     * On success will return 'success', and on failure will return an error message
+     *
+     * @param string $repoUrl
```

Comment:
Is there a design reason that on both success and failure, a string is returned? Wouldn't it be more conventional if this function returned a boolean, or a wrapped Response instance?

Reply:
On failure we want to convey why it failed, so a boolena would not be sufficient. Since we are at the gateway level, we wouldn't really be returning a response instance. It could be wrapped in a Data Transfer Object (DTO) or Entity, but I felt it was kinda overkill in this case because I don't ever see a case where we would need to return anything other than an error message. In the case of a success, the error message would be success. I suppose on success we could return the created entity... That might actually be better. What do you think?

Reply:
Actually, all of the changes I need for this are in a later PR, because I think I changed the way I did some of this stuff... Sooo. Yeah.

Reply:
I don't have the infrastructure in this PR, and it is created in a later one. So I'll add it in a later one where it makes more sense. This PR didn't target that as functionality. It has been done in PR #105 on commit 94c5e8e